litmus-portal/graphql-server/graph/schema.resolvers.go

```diff
@@ -407,14 +418,32 @@ func (r *queryResolver) GetHubStatus(ctx context.Context, projectID string) ([]*
 }
 
 func (r *queryResolver) GetYAMLData(ctx context.Context, experimentInput model.ExperimentInput) (string, error) {
+	err := authorization.ValidateRole(ctx, experimentInput.ProjectID,
+		authorization.MutationRbacRules[authorization.GetYAMLData],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return "", err
+	}
 	return myhub.GetYAMLData(ctx, experimentInput)
 }
 
 func (r *queryResolver) GetPredefinedWorkflowList(ctx context.Context, hubName string, projectID string) ([]string, error) {
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.PredefinedWorkflowOperations],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return nil, err
+	}
 	return myhub.GetPredefinedWorkflowList(hubName, projectID)
 }
 
 func (r *queryResolver) GetPredefinedExperimentYaml(ctx context.Context, experimentInput model.ExperimentInput) (string, error) {
+	err := authorization.ValidateRole(ctx, experimentInput.ProjectID,
+		authorization.MutationRbacRules[authorization.PredefinedWorkflowOperations],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return "", err
+	}
```

Comment:
Make sure the ProjectIDs passed here are actually used to get the required data else, if they are just redundant information the same attack vector exists here too.

Reply:
@gdsoumya  For the above 2 cases, the Project ID is being used to fetch the chaos-charts data from the hub directories. I believe it wont cause any issue here. Have updated the code for other cases.

Reply:
Awesome thanks.